### PR TITLE
Update neo-go version

### PR DESCRIPTION
**Pull request by fyrchik**
Related #49 .
Need to wait until preview4 to update C# nodes.

### CODE REVIEW

**Comment by roman-khimov** [comment]
>  Need to wait until preview4 to update C# nodes.

I don't think so, they should work fine with latest NEP-17 PRs merged. The trick here is exactly to make it all work as of current state of both nodes (and they're relatively synchronized at the moment).

**Comment by fyrchik** [comment]
@roman-khimov the problem is that C# docker file requires release tarball, and I haven't found if built for master.

**Comment by roman-khimov** [comment]
See Dockerfile.sharp.sources.from_binaries (and Dockerfile.sharp.sources.from_local_dependencies, though that might be more involved, it's OK to have just the first one for now).

**Comment by fyrchik** [comment]
There are two unmerged PRs https://github.com/neo-project/neo-node/pull/681 https://github.com/neo-project/neo/pull/2083 . Because of incompatible container fails to be built. I'll try using `git` instead of downloading zip files.

**Comment by roman-khimov** [comment]
Just leave neo-vm at compatible commit, it hasn't been changed functionally for quite some time and we certainly don't need .NET 5.0 at the moment.

**Comment by fyrchik** [comment]
Updated `neo` to commit right before introduction of management contract and `neo-vm` right before dotnet 5.0 update. Everything builds, though nodes accepted by C# node are empty. Nothing is logged, currently investigating.

**Comment by fyrchik** [comment]
Both nodes build and run bench. But C# node accepts only 512 tx per block. Seems like policy contract invoke somehow failed.

File: .docker/ir/docker-compose.single.sharp.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
       - stats
     image: registry.nspcc.ru/neo-bench/neo-sharp:bench
     logging:
-      driver: "none"
+      driver: "journald"
```
Comment (roman-khimov) [inline .docker/ir/docker-compose.single.sharp.yml:9]: NACK.

File: Makefile [hunk 2]
```diff
@@ -13,9 +13,9 @@ DC_SHARP_RPC=.docker/rpc/docker-compose.sharp.yml
 
 DF_GO=.docker/build/Dockerfile.golang
 DF_BENCH=.docker/build/Dockerfile.bench
-DF_SHARP=.docker/build/Dockerfile.sharp
+#DF_SHARP=.docker/build/Dockerfile.sharp
 #DF_SHARP=.docker/build/Dockerfile.sharp.sources.from_binaries
```
Comment (roman-khimov) [inline Makefile:17]: I thought this one is easier to make work actually.

**Comment by roman-khimov** [comment]
> But C# node accepts only 512 tx per block.

Well, why is that?

**Comment by fyrchik** [comment]
The problem is in invalid policy contract hash (on a commit previous to management contract introduction).

**Comment by fyrchik** [comment]
The problem was in contract hashes. Commit for `neo-go` with fix is from https://github.com/nspcc-dev/neo-go/pull/1616 . Another thing is increased cost of `transfer` (TBD fix in `neo-go`).

File: .docker/build/Dockerfile.sharp.sources.from_local_dependencies [hunk 3]
```diff
@@ -20,11 +20,11 @@ RUN set -x \
...
 #           └── neo-vm
 # Use ENV variables to specify branch, version or revision to download.
-ENV CLIBRANCH="cc7307c5eef5c7f77772c037f4b7619dae55a6bb"
-ENV MODULESBRANCH="e8a69c460be235cd9f10de3d1a2667b150252a52"
+ENV CLIBRANCH="92363c5e66bd6028f73c53ab77492a11c783262e"
+ENV MODULESBRANCH="c995e65164ae10a303ec7515af5a0e93dc493be2"
 ENV MODULES="LevelDBStore RocksDBStore RpcServer"
 ENV NEOVMBRANCH="a14b934f5b5c8779164e3b96b276984b3a57ccd4"
-ENV NEOBRANCH="54eff7e3662085a86a5e6cfefdc3e019215830f4"
+ENV NEOBRANCH="2e6350136dd9cd36cc0804845b2eedd75a053a7f"
```
Comment (AnnaShaleva) [inline .docker/build/Dockerfile.sharp.sources.from_local_dependencies:27]: If building from local dependencies works, then building from binaries should work too. Could you also update Dockerfile.sharp.sources.from_binaries?

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

File: .docker/build/Dockerfile.sharp.sources.from_local_dependencies (hunk 3 above)
Comment (fyrchik) [reply .docker/build/Dockerfile.sharp.sources.from_local_dependencies]: Fixed.

**Comment by fyrchik** [comment]
Updated `neo-go` and `neo` to master, `neo-vm` is still one commit behind `.NET` update.

File: .docker/build/Dockerfile.golang [hunk 4]
```diff
@@ -6,7 +6,7 @@ WORKDIR /neo-go
 # Install deps:
 RUN apk add --no-cache git
 
-ARG REV="5f21178fe958df1afa8047c341257a2f3218a167"
+ARG REV="44b4c929922240a91a535c054da6f071f6c7ad64"
```
Comment (AnnaShaleva) [inline .docker/build/Dockerfile.golang:9]: Could you, please, update Go revision to ecdf95d5d6d8fc648ebbece74f8294e62201b765? It is more recent than 44b4c929922240a91a535c054da6f071f6c7ad64 and contains mempool bugfix which is important for bench.

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]
If all the logic changes needed are in place, we'd rather merge it as is to unblock #52. Changing neo-go revision is rather trivial.

**Pull request merged**